### PR TITLE
Do not append `mlflow.runName` tag if it's already in tags

### LINKED ISSUE #7133: [BUG] 1.30: mlflow.exceptions.RestException: BAD_REQUEST: (sqlite3.IntegrityError) UNIQUE constraint failed: tags.key, tags.run_uuid
### Issues Policy acknowledgement

- [X] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Willingness to contribute

No. I cannot contribute a bug fix at this time.

### MLflow version

1.30

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: ubuntu 20.04
- **Python version**: 3.9.15
- **yarn version, if running the dev UI**:

### Describe the problem

After starting version 1.30 via:

mlflow server -h 0.0.0.0 -p 5050 \
  --backend-store-uri sqlite:////home/mlflow/mlflow.db \
  --serve-artifacts \
  --artifacts-destination file:/home/mlflow/mlartifacts

on a fresh install, I get:

mlflow.exceptions.RestException: BAD_REQUEST: (sqlite3.IntegrityError) UNIQUE constraint failed: tags.key, tags.run_uuid

### Tracking information

>>> import mlflow
>>> print("MLflow version:", mlflow.__version__)
MLflow version: 1.30.0
>>> print("Tracking URI:", mlflow.get_tracking_uri())
Tracking URI: file:///home/mlflow/mlruns
>>> print("Artifact URI:", mlflow.get_artifact_uri())
Artifact URI: file:///home/mlflow/mlruns/0/78c1349a05df4b7aada9a7f05bccd26e/artifacts

### Code to reproduce issue

with mlflow.start_run(**kwargs) as run:
  ...

### Stack trace

  File "/home/kaina/work/repo/core/scripts/tfx/mlflow_continuation_example.py", line 151, in do_run
    with mlflow.start_run(**kwargs) as run:
  File "/home/kaina/.pyenv/versions/core/lib/python3.9/site-packages/mlflow/tracking/fluent.py", line 318, in start_run
    active_run_obj = client.create_run(experiment_id=exp_id_for_run, tags=resolved_tags)
  File "/home/kaina/.pyenv/versions/core/lib/python3.9/site-packages/mlflow/tracking/client.py", line 265, in create_run
    return self._tracking_client.create_run(experiment_id, start_time, tags)
  File "/home/kaina/.pyenv/versions/core/lib/python3.9/site-packages/mlflow/tracking/_tracking_service/client.py", line 99, in create_run
    return self.store.create_run(
  File "/home/kaina/.pyenv/versions/core/lib/python3.9/site-packages/mlflow/store/tracking/rest_store.py", line 170, in create_run
    response_proto = self._call_endpoint(CreateRun, req_body)
  File "/home/kaina/.pyenv/versions/core/lib/python3.9/site-packages/mlflow/store/tracking/rest_store.py", line 56, in _call_endpoint
    return call_endpoint(self.get_host_creds(), endpoint, method, json_body, response_proto)
  File "/home/kaina/.pyenv/versions/core/lib/python3.9/site-packages/mlflow/utils/rest_utils.py", line 256, in call_endpoint
    response = verify_rest_response(response, endpoint)
  File "/home/kaina/.pyenv/versions/core/lib/python3.9/site-packages/mlflow/utils/rest_utils.py", line 185, in verify_rest_response
    raise RestException(json.loads(response.text))
mlflow.exceptions.RestException: BAD_REQUEST: (sqlite3.IntegrityError) UNIQUE constraint failed: tags.key, tags.run_uuid
[SQL: INSERT INTO tags ("key", value, run_uuid) VALUES (?, ?, ?)]

### Other info / logs

_No response_

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [X] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

Fix #7133

## How is this patch tested?

- [x] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by BenWilson2 — APPROVED** [review]
LGTM!

File: mlflow/store/tracking/file_store.py [hunk 1]
```diff
@@ -604,7 +604,17 @@ def create_run(self, experiment_id, user_id, start_time, tags, run_name):
                 "Could not create run under non-active experiment with ID %s." % experiment_id,
                 databricks_pb2.INVALID_STATE,
             )
-        run_name = run_name if run_name else _generate_random_name()
+        tags = tags or []
+        run_name_tag = _get_run_name_from_tags(tags)
+        if run_name and run_name_tag:
+            raise MlflowException(
+                "Both 'run_name' argument and 'mlflow.runName' tag (deprecated) are specified. "
```
Comment (harupy) [inline mlflow/store/tracking/file_store.py:611]:
```suggestion
                "Both 'run_name' argument and 'mlflow.runName' tag are specified. "
```

File: mlflow/store/tracking/sqlalchemy_store.py [hunk 2]
```diff
@@ -548,7 +548,17 @@ def create_run(self, experiment_id, user_id, start_time, tags, run_name):
             artifact_location = append_to_uri_path(
                 experiment.artifact_location, run_id, SqlAlchemyStore.ARTIFACTS_FOLDER_NAME
             )
-            run_name = run_name if run_name else _generate_random_name()
+            tags = tags or []
+            run_name_tag = _get_run_name_from_tags(tags)
+            if run_name and run_name_tag:
+                raise MlflowException(
+                    "Both 'run_name' argument and 'mlflow.runName' tag (deprecated) are specified. "
```
Comment (harupy) [inline mlflow/store/tracking/sqlalchemy_store.py:555]:
```suggestion
                    "Both 'run_name' argument and 'mlflow.runName' tag are specified. "
```

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @harupy !

File: tests/store/tracking/test_file_store.py [hunk 3]
```diff
@@ -819,6 +820,30 @@ def test_create_run_sets_name(self):
...
+            run_name=None,
+            tags=[RunTag(MLFLOW_RUN_NAME, "test")],
+        ).info.run_id
+        run = fs.get_run(run_id)
+        assert run.info.run_name == "test"
+
+        with pytest.raises(
+            MlflowException,
+            match=re.escape(
+                "Both 'run_name' argument and 'mlflow.runName' tag (deprecated) are specified."
```
Comment (harupy) [inline tests/store/tracking/test_file_store.py:836]:
```suggestion
                "Both 'run_name' argument and 'mlflow.runName' tag are specified."
```

File: tests/store/tracking/test_sqlalchemy_store.py [hunk 4]
```diff
@@ -906,6 +907,29 @@ def test_create_run_sets_name(self):
...
+            run_name=None,
+            tags=[RunTag(mlflow_tags.MLFLOW_RUN_NAME, "test")],
+        ).info.run_id
+        run = self.store.get_run(run_id)
+        assert run.info.run_name == "test"
+
+        with pytest.raises(
+            MlflowException,
+            match=re.escape(
+                "Both 'run_name' argument and 'mlflow.runName' tag (deprecated) are specified."
```
Comment (harupy) [inline tests/store/tracking/test_sqlalchemy_store.py:923]:
```suggestion
                "Both 'run_name' argument and 'mlflow.runName' tag are specified."
```

**Comment by harupy** [comment]
@mlflow-automation autoformat

**Comment by bveeramani** [comment]
Hey @harupy, do you happen to know if this fix will be included in version `1.30.1`? If so, do you know when `1.30.1` will be released? The bug is causing issues with Ray's MLflow integration: https://github.com/ray-project/ray/issues/29749.

**Comment by harupy** [comment]
@bveeramani The fix will be included in MLflow 2.0.0 (which will be released soon).

**Comment by bveeramani** [comment]
@harupy thanks! That's good to know.

**Pull request merged**